cloud/scope/managed_machine_pool.go

```diff
@@ -343,6 +349,67 @@ func (m *ManagedMachinePoolScope) CreateNodePool(ctx context.Context) (*oke.Node
 	return m.getOKENodePoolFromOCID(ctx, nodePoolId)
 }
 
+func (m *ManagedMachinePoolScope) setNodepoolImageId(ctx context.Context) error {
+	imageId := m.OCIManagedMachinePool.Spec.NodeSourceViaImage.ImageId
+	if imageId != nil && *imageId != "" {
+		return nil
+	}
+	response, err := m.ContainerEngineClient.GetNodePoolOptions(ctx, oke.GetNodePoolOptionsRequest{
+		NodePoolOptionId: common.String("all"),
+		CompartmentId:    common.String(m.OCIManagedCluster.Spec.CompartmentId),
+	})
+	if err != nil {
+		m.Logger.Error(err, "Could not lookup node pool options")
+		return err
+	}
+	// version in the capoci spec starts with v, wheres as in the node pool options API in OKE
+	// the image name does not contain . For example
+	//       {
+	//        "image-id": "ocid1.image.oc1.iad.aaaaaaaafp7ysdbfl2bg67s4jzbpewxo3k772baixa3vwzecwogl474qecza",
+	//        "source-name": "Oracle-Linux-8.6-aarch64-2022.12.15-0-OKE-1.25.4-543",
+	//        "source-type": "IMAGE"
+	//      },
+	//      {
+	//        "image-id": "ocid1.image.oc1.iad.aaaaaaaauwwokidwf5nfi34ucbvgnjsni3klnfmu6pz73ctdohb2byiw6ztq",
+	//        "source-name": "Oracle-Linux-8.6-2022.12.15-0-OKE-1.23.4-543",
+	//        "source-type": "IMAGE"
+	//      },
+	// we will only default to non gpu image
+	k8sVersion := (*m.OCIManagedMachinePool.Spec.Version)[1:]
+	shape := m.OCIManagedMachinePool.Spec.NodeShape
+	isArmShape := strings.Contains(shape, "A1")
+	for _, source := range response.Sources {
+		image, ok := source.(oke.NodeSourceViaImageOption)
+		if ok {
+			sourceName := *image.SourceName
+			if isValidImage(sourceName) {
+				// if source is an arm source and expectation is not an arm image, ignore
+				if strings.Contains(sourceName, "aarch64") && !isArmShape {
+					continue
+				}
+				if strings.Contains(sourceName, k8sVersion) {
+					m.Info("Image being used", "Name", sourceName, "OCID", *image.ImageId)
+					m.OCIManagedMachinePool.Spec.NodeSourceViaImage.ImageId = image.ImageId
+					return nil
+				}
+			}
+		}
+	}
```

Comment:
The above for loop as many possible failure modes. Should we be logging any of them?

For example: do we care if the `source` can't be converted and `ok` is false?

I'm just wondering if `"could not lookup nodepool image id from nodepool options"` is enough context to address any issues users might run into.

Reply:
added the entire options response to the logs